modules/util/url.go

```diff
@@ -28,14 +27,21 @@ func URLJoin(base string, elems ...string) string {
 	if err != nil {
 		return ""
 	}
-	joinedPath := path.Join(elems...)
-	argURL, err := url.Parse(joinedPath)
-	if err != nil {
-		return ""
+
+	var fragment string
+	last := len(elems) - 1
+	if len(elems) > 0 {
+		if strings.Contains(elems[last], "#") {
+			elems[last], fragment, _ = strings.Cut(elems[last], "#")
```

Comment:
banch name can also have # like `BranchName#123`

Reply:
That's correct but does not affect the method.

![grafik](https://github.com/go-gitea/gitea/assets/1666336/ad57a071-ab24-46b6-ad54-107d65a80e13)
The screenshot shows a branch with `#` with a readme which includes the picture from the repo.

Reply:
cool, works for me then.

Reply:
Some test cases are failing 
<img width="414" alt="image" src="https://github.com/go-gitea/gitea/assets/80308335/fbe1fb04-d9a3-41b7-86c3-598496a4a008">

Reply:
That's why this PR is marked as depending on the other PR. If #26745 is merged, the test will work.

Reply:
Ohk,  got it. my bad.